### ISSUE TITLE
Can Java requirement be specified more prominently?

### ISSUE THREAD

**Issue by kwalcock** [softwaremill/sttp#1982]
With Java 8, I get the exception below *at runtime* but it goes away with Java 11.  Would I need to resort to a different back end?
```
Exception in thread "main" java.lang.BootstrapMethodError: java.lang.NoClassDefFoundError: java/net/http/HttpRequest
	at sttp.client4.DefaultSyncBackend$.apply(DefaultSyncBackend.scala:12)
	at sttp.client4.quick$.backend$lzycompute(quick.scala:4)
	at sttp.client4.quick$.backend(quick.scala:4)
	at sttp.client4.quick$RichRequest.send(quick.scala:7)
        ...

Caused by: java.lang.ClassNotFoundException: java.net.http.HttpRequest
	at java.net.URLClassLoader.findClass(URLClassLoader.java:381)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	... 31 more
```

**Comment by adamw** [comment]
The Java 11 requirement is specified here: https://sttp.softwaremill.com/en/stable/quickstart.html, as well as in the main docs of version 4: https://sttp.softwaremill.com/en/latest/

The default backend in sttp client is based on `HttpClient` from the JDK, which is available from Java 11+. I think trying another backend might work, but there might be other problems as well.

**Comment by kwalcock** [comment]
Thanks.  I didn't check hard enough :-(  because it didn't occur to me to be a possibility.  I stopped at Scala 2.12, thinking that it implied Java 8 compatibility and then followed the sbt libraryDependency instructions.  My mistake.  Technically, "supported Java versions include 11+" does not rule out support for Java 8, so perhaps the statement can be strengthened.

**Comment by adamw** [comment]
@kwalcock So it doesn't work using a different backend as well?

**Comment by kwalcock** [comment]
I haven't tried it yet.

**Comment by pvgoran** [comment]
I was successfully using Sttp 3.7.x with Java 8, using `AsyncHttpClientZioBackend`. When I tried to upgrade to 3.8.x, it stopped working, **at least when an exception was thrown during request**: I got a ClassNotFoundException for `java.net.http.HttpTimeoutException`. The commit 21393d202cf85e2c12b969b1820beee634113dba introduced `java.net.http.HttpTimeoutException` into Sttp code, after this commit, `SttpClientExceptionExtensions` can't be loaded by Java 8 because `java.net.http.HttpTimeoutException` doesn't exist in Java 8.

Long story short: Sttp 3.8.3 seems to be usable with Java 8 (if the `AsyncHttpClientZioBackend` backend is used), Sttp 3.8.4 is unusable with Java 8.
